alpine-server/src/main/java/alpine/server/auth/ApiKeyAuthenticationService.java

```diff
@@ -37,12 +37,16 @@ public class ApiKeyAuthenticationService implements AuthenticationService {
 
     /**
      * Given the specified ContainerRequest, the constructor retrieves a header
-     * named 'X-Api-Key', if it exists.
+     * named 'X-Api-Key' or a URI query parameter named 'apiKey', if they exist.
      * @param request the ContainerRequest object
      * @since 1.0.0
      */
-    public ApiKeyAuthenticationService(final ContainerRequest request) {
-        this.assertedApiKey = request.getHeaderString("X-Api-Key");
+    public ApiKeyAuthenticationService(final ContainerRequest request, boolean byQuery) {
+        if (byQuery) {
+            this.assertedApiKey = request.getUriInfo().getQueryParameters().getFirst("apiKey");
+        } else {
+            this.assertedApiKey = request.getHeaderString("X-Api-Key");
+        }
```

Comment:
I think this shouldn't be either-or, but instead based on priority, i.e.

1. If `X-Api-Key` header exists, take its value
2. If sourcing from query parameter is allowed, and `apiKey` parameter exists, take its value
3. Otherwise, take `null`

I like that you added an annotation to whitelist usage via query parameter. That is good, and makes it explicit when it's allowed. But it shouldn't disallow authentication via the usual `X-Api-Key` header.

Reply:
Done that